.ci/build_demo.sh

```diff
@@ -0,0 +1,150 @@
+    echo -e "${BLUE}Use only lowercase letters and numbers, you can include dashes and dots in the middle (eg. \"example.com\").${NC}"
+    exit 1
+fi
+
+echo -e "${BLUE}Configuring your demo-shop...${NC}"
+
+# Configuration files are baked into the Docker image at the moment
+cp "$WORKSPACE/app/config/parameters.yml.dist" "$WORKSPACE/app/config/parameters.yml"
+cp "$WORKSPACE/app/config/domains_urls.yml.dist" "$WORKSPACE/app/config/domains_urls.yml"
+
+# Disable master e-mail and mailer whitelist
+sed -i "s/mailer_master_email_address:.*/mailer_master_email_address: ~/" "$WORKSPACE/app/config/parameters.yml"
+sed -i "s/mailer_delivery_whitelist:.*/mailer_delivery_whitelist: ~/" "$WORKSPACE/app/config/parameters.yml"
+
+# Fetch all domain IDs
+DOMAIN_IDS=$(cat "$WORKSPACE/app/config/domains_urls.yml" | grep -Po 'id: ([0-9]+)$' | sed -r 's/id: ([0-9]+)/\1/')
+
+# Modify public URLs to $DOMAIN_ID.$DEMO_NAME.$DEMO_DOMAIN ($DOMAIN_ID is omitted for the first domain)
+for DOMAIN_ID in $DOMAIN_IDS; do
+    if [[ "$DOMAIN_ID" == "1" ]]; then
+        # 1st domain has URL without a number prefix
+        sed -i "/id: 1/,/url:/{s/url:.*/url: http:\/\/$DEMO_NAME.$DEMO_DOMAIN/}" "$WORKSPACE/app/config/domains_urls.yml"
+    else
+        # 2nd and subsequent domains have URLs with DOMAIN_ID prefix
+        sed -i "/id: $DOMAIN_ID/,/url:/{s/url:.*/url: http:\/\/$DOMAIN_ID.$DEMO_NAME.$DEMO_DOMAIN/}" "$WORKSPACE/app/config/domains_urls.yml"
+    fi
+done
+
+echo -e "${BLUE}Connecting to shared elasticsearch instance...${NC}"
+
+NETWORK_CONNECT='false'
+
+if docker network create "$ELASTICSEARCH_NETWORK" &> /dev/null; then
+    echo "A network $ELASTICSEARCH_NETWORK has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The network $ELASTICSEARCH_NETWORK already exists."
+fi
+
+if [[ -z "$(docker ps -q -f name="$ELASTICSEARCH_CONTAINER")" ]]; then
+    echo "Creating a shared elasticsearch container $ELASTICSEARCH_CONTAINER..."
+    docker run --name "$ELASTICSEARCH_CONTAINER" --ulimit "nofile=65536:65536" --env "discovery.type=single-node" -d "docker.elastic.co/elasticsearch/elasticsearch-oss:6.3.2"
+    echo "The elasticsearch container has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The elasticsearch container $ELASTICSEARCH_CONTAINER already exists."
+fi
+
+if [[ "$NETWORK_CONNECT" == 'true' ]]; then
+    echo "Connecting the container to the network...."
+    docker network connect "$ELASTICSEARCH_NETWORK" "$ELASTICSEARCH_CONTAINER"
+fi
+
+ELASTICSEARCH_IP=$(docker inspect -f "{{ with (index .NetworkSettings.Networks \"$ELASTICSEARCH_NETWORK\") }}{{ .IPAddress }}{{ end }}" "$ELASTICSEARCH_CONTAINER")
+if [[ -z "$ELASTICSEARCH_IP" ]]; then
+    echo -e "${RED}The IP address of the elasticsearch container inside the shared network could not be obtained!${NC}"
+    exit 1
+else
+    echo -e "The IP address of ${BLUE}$ELASTICSEARCH_CONTAINER${NC} inside ${BLUE}$ELASTICSEARCH_NETWORK${NC} is ${BLUE}$ELASTICSEARCH_IP${NC}"
```

Comment:
During testing, this was pretty important information which basically said that everything is understood and parsed correctly by the script. In case of any errors, it may provide very useful hints. But I do not object if you want to omit this.